app/models/notification.rb

```diff
@@ -25,7 +25,11 @@ def type=(typ)
   def self.emit_for_event(event)
     return unless NotificationType.names.include?(event.event_type)
     type = NotificationType.find_by(:name => event.event_type)
-    Notification.create(:notification_type => type, :subject => event.target)
+    return if type.none?
+    Notification.create(:notification_type => type,
+                        :options           => event.full_data,
+                        :subject_id        => event.target_id,
+                        :subject_type      => event.target_type)
```

Comment:
Do subject `id` and `type` need to be broken out or can it remain as it was before `:subject => event.target`

Reply:
I liked the idea of not looking up the target object, but this is fine